test/unit/lbp-wrapper-factory.js

```diff
@@ -0,0 +1,150 @@
+const {expect} = require('chai');
+const { ethers } = require("hardhat");
+const { parseEther } = ethers.utils;
+
+const init = require("../test-init.js");
+const balancer = require("../helpers/balancer.js");
+const tokens = require("../helpers/tokens.js");
+const {constants, BN, expectRevert} = require('@openzeppelin/test-helpers');
+
+const deploy = async () => {
+	const setup = await init.initialize(await ethers.getSigners());
+	
+	setup.vault = await balancer.Vault(setup);
+	
+	setup.lbpFactory = await balancer.LBPFactory(setup);
+
+    setup.lbpWrapper = await init.getDeployedContract(
+		"LBPWrapper", setup.roles.root);
+
+	setup.Lbp = balancer.Lbp(setup);
+
+	setup.tokenList = await tokens.ERC20TokenList(2, setup.roles.root);
+
+	return setup;
+};
+
+function sortTokens(tokens) {
+	if (tokens[0].address > tokens[1].address) {
+		const temp = tokens[0];
+		tokens[0] = tokens[1];
+		tokens[1] = temp;
+	};
+
+	return tokens;
+};
+
+describe(">> Contract: WrapperFactory", () => {
+	let setup, swapsEnabled;
+	let tokenAddresses, admin, owner, sortedTokens, newOwner;
+
+	const startTime = Date.now();
+	const endTime = startTime+100000;
+	
+	const NAME = "SEED-MKR POOL";
+	const SYMBOL = "SEED-MKR";
+	
+	const START_WEIGHTS = [0.7e18, 0.3e18].map((weight) => weight.toString());
+	const END_WEIGHTS = [0.3e18, 0.7e18].map((weight) => weight.toString());
+	const ADMIN_BALANCE = [32.667e18, 30000e6].map((balance) => balance.toString());
+
+	const AMOUNTS = [16.667e18, 15000.e6].map((amount) => amount.toString());
+	const SWAP_FEE_PERCENTAGE = 0.5e16.toString(); // 0.5%
+	const JOIN_KIND_INIT = 0;
+	const ZERO_ADDRESS = constants.ZERO_ADDRESS
+	30000000000
+
+	context("» deploy LBP WrapperFactory", () => {
+		before("!! setup", async () => {
+			setup = await deploy();
+			
+			swapsEnabled = true;
+
+			({ root: owner, prime: admin,  beneficiary: newOwner} = setup.roles);
+			sortedTokens = sortTokens(setup.tokenList);
+			// Need to solve this in tokens.js helper file for > 2 tokens.
+			tokenAddresses = sortedTokens.map((token) => token.address);
+
+		});
```

Comment:
I'm not a big fan of using `before` to initialize variables that will be manipulated during the tests. I think it doesn't go well with the paradigm of [Arrange-Act-Assert](https://medium.com/swlh/aaa-the-bread-and-butter-of-testing-b31d99c30aca), because in later `it` tests it becomes blurrish what had been "arranged" in previous `it` blocks. But I guess this is up for discussion.